assets/aml-benchmark/scripts/custom_inference_postprocessors/humaneval.py

```diff
@@ -184,17 +184,24 @@ def run_humaneval_postprocessor(
     # Post processing the prediction and ground truth columns
     for row in pred_dict_full:
         gt = "\n" + row["test"] + "\n" + "check(" + row["entry_point"] + ")"
-        if str("def " + row["entry_point"] + "(") in row["original_prediction"]:
-            # If the model regenerates the prompt/ function name
-            pred_combined_prompt = _extract_text_from_markdown_tag(row["original_prediction"], tag_type='python')
-        else:
-            original_prediction = _extract_text_from_markdown_tag(row["original_prediction"], tag_type='python')
+        tag_type = "python" if "python" in row["original_prediction"] else ""
+
+        # Extract the prediction data from the markdown tags
+        pred_combined_prompt = _extract_text_from_markdown_tag(row["original_prediction"], tag_type)
+
+        # Get the index of the first function definition and return keyword
+        func_name_index = pred_combined_prompt.find(str("def " + row["entry_point"] + "("))
+        return_keyword_index = pred_combined_prompt.find("return")
```

Comment:
I think we need a different logic. When the input partial code itself contains one function which has a `return` statement, will the current code work?

Reply:
Can you give an example?

Reply:
Handled this scenario. Example details here 
[HumanEval_38.txt](https://github.com/Azure/azureml-assets/files/15020692/HumanEval_38.txt)